pkg/controller/synchronize.go

```diff
@@ -81,16 +85,26 @@ func Synchronize(opts *SynchronizeOpts) error {
 }
 
 // handleNode knows how to run Reconcile() on every node of a ResourceNode tree
-func handleNode(reconcilerManager reconciler.Reconciler, currentNode *resourcetree.ResourceNode) error {
-	_, err := reconcilerManager.Reconcile(currentNode)
-	if err != nil {
-		return fmt.Errorf("error reconciling node: %w", err)
+func handleNode(reconcilerManager reconciler.Reconciler, currentNode *resourcetree.ResourceNode) (err error) {
+	reconciliationResult := reconciler.ReconcilationResult{Requeue: true, RequeueAfter: 0 * time.Second}
+
+	for requeues := 0; reconciliationResult.Requeue; requeues++ {
```

Comment:
Maybe we should replace this with an exponential backoff approach: https://github.com/cenkalti/backoff/, the default values for this library might make sense to drop in as are: https://pkg.go.dev/github.com/cenkalti/backoff/v4#pkg-constants. Worth considering at least.

Reply:
I don't know anything about that package. I feel we get everything we need from the functionality implemented. Might want to add a multiplier at some point for exponential increase in delay time, but thats out of scope for this PR for now. Feel free to add it if you want, but I really want this merged before nightly tonight :P